test-data/unit/check-classes.test

```diff
@@ -894,6 +892,295 @@ a.f = a.f # E: Property "f" defined in "A" is read-only
 a.f.x # E: "int" has no attribute "x"
 [builtins fixtures/property.pyi]
 
+-- Descriptors
+-- -----------
+
+
+[case testAccessingNonDataDescriptor]
+class D:
+    def __get__(self, instance: Any, owner: Any) -> str:
+        return 'some string'
+
+class A:
+    f = D()
+
+a = A()
+reveal_type(a.f)  # E: Revealed type is 'builtins.str'
+
+[case testSettingNonDataDescriptor]
+class D:
+    def __init__(self, value: str) -> None:
+        self.value = value
```

Comment:
To reduce the test size you can omit `__init__` and make `__get__` return a literal.